package/MDAnalysis/coordinates/memory.py

```diff
@@ -346,15 +400,26 @@ def parse_n_atoms(filename, order='fac', **kwargs):
 
     def copy(self):
         """Return a copy of this Memory Reader"""
+        vels = (self.velocity_array.copy()
+                if self.velocity_array is not None
+                else None)
+        fors = (self.force_array.copy()
+                if self.force_array is not None
+                else None)
+        dims = (self.dimensions_array.copy()
+                if self.dimensions_array is not None
+                else None)
         new = self.__class__(
             self.coordinate_array.copy(),
             order=self.stored_order,
-            dimensions=self.ts.dimensions,
+            dimensions=dims,
+            velocities=vels,
+            forces=fors,
             dt=self.ts.dt,
             filename=self.filename,
         )
         new[self.ts.frame]
-        new.ts = self.ts.copy()
```

Comment:
this line was problematic as `base.Timestep.copy()` isn't respectful of MemoryReader behaviour, see #2081.  I don't think removing it causes any problems, as `new[self.ts.frame]` should populate the Timestep with everything needed